OpenDreamShared/Compiler/DM/DMParser.cs

```diff
@@ -418,13 +418,42 @@ public class DMParser : Parser<Token> {
                 Whitespace();
                 DMASTPath path = Path();
                 if (path == null) Error("Expected a variable name");
-                Whitespace();
 
                 DMASTExpression value = null;
 
+                if (Check(TokenType.DM_LeftBracket)) //TODO: Multidimensional lists
+                {
+                    //Type information
+                    if (path.Path.PathString.Substring(0, 4) != "list")
+                    {
+                        path = new DMASTPath(new DreamPath("list/"  + path.Path.PathString));
+                    }
+
+                    var size = Expression() as DMASTExpressionConstant;
+                    if (size is not DMASTConstantInteger)
```

Comment:
After some testing, it appears DM will actually take any expression as a valid size for the list (including non-constants). The value of the variable is also `null` when the brackets are empty (`var/list/L[]`).